htdocs/contrat/class/contrat.class.php

```diff
@@ -774,6 +719,7 @@ function fetch_lines()
 				$line->fetch_optionals($line->id,$extralabelsline);
 
 				$this->lines[]			= $line;
+				$this->lines_id_index_mapper[$line->id] = key($this->lines);
```

Comment:
Why storing in a second property  lines_id_index_mapper ?
Why not just doing 

$this->lines[$line->id] = $line;